src/metabase/search/task/search_index.clj

```diff
@@ -40,10 +40,10 @@
 (task/defjob ^{DisallowConcurrentExecution true
                :doc                        "Populate a new Search Index"}
   SearchIndexReindex [_ctx]
-  (search/reindex!))
+  (.get ^Future (search/reindex!)))
```

Comment:
TIL that you can `@java-future` - is there a reason you're using interop instead?